### PR TITLE
[Custom DC] Determine root import directory on the fly and use that as import group name

### PR DESCRIPTION
**Pull request by Fructokinase**

### CODE REVIEW

File: bigtable_automation/gcf/gcs.go [hunk 1]
```diff
@@ -104,7 +104,12 @@ func GenerateManifest(ctx context.Context, bucket, pathToDataFolder string) (*pb
 		return nil, err
 	}
 
-	importGroupName := "custom"
+	// IMPORTANT NOTE: importGroupName has a character limit of 21
```
Comment (shifucun) [inline bigtable_automation/gcf/gcs.go:107]: why has this limit?

File: bigtable_automation/gcf/controller_trigger.go [hunk 2]
```diff
@@ -102,3 +103,30 @@ func (s CustomDCPubSubMsg) Publish(ctx context.Context, p PublishConfig) error {
...
+// Find the root directory for data imports on gcs.
+// For data import directory structures, please refer to
+// https://docs.datacommons.org/custom_dc/upload_data.html
+//
+// Root directory in the above example is "gcs_folder"
+// Example:
+// findRootImportDirectory("/somepath/gcs_folder/internal/control/trigger.txt")
+// returns "/somepath/gcs_folder/"
+//
+// Note manifest generator requires the "/" at the end.
```
Comment (shifucun) [inline bigtable_automation/gcf/controller_trigger.go:116]: Maybe start to add test, can use mock GCS as suggested in https://stackoverflow.com/questions/39814108/how-to-unit-test-google-cloud-storage-in-golang

Comment (Fructokinase) [reply bigtable_automation/gcf/controller_trigger.go:116]: Will leave this as a todo

File: bigtable_automation/gcf/gcs.go (hunk 1 above)
Comment (Fructokinase) [reply bigtable_automation/gcf/gcs.go:107]: > why has this limit?

This is our internal limit

**Review by shifucun — APPROVED** [review]

**Pull request merged**